draft-pauly-intarea-proxy-config-pvd.md

```diff
@@ -137,15 +137,40 @@ match the hostname of the HTTP proxy. The "prefixes" array SHOULD be empty by de
 # Enumerating proxies within a PvD {#proxy-enumeration}
 
 This document defines a new PvD Additional Information key, `proxies`, that
-is an array of strings that is a list of proxy URIs (or URI templates
-{{!URITEMPLATE=RFC6570}}) that are available as part of a PvD. The new
-key is registered in {{iana}}.
-
-The kind of proxy is implied by the URI scheme and any template variables.
-For example, since UDP proxying {{CONNECTUDP}} has the URI template variables
-`target_host` and `target_port`, the URI
-"https://proxy.example.org:4443/masque{?target_host,target_port}" implies
-that the proxy supports UDP proxying.
+is an array of dictionaries, where each dictionary in the array defines
+a single proxy that is available as part of the PvD (see {{proxies-key-iana}}).
+Each proxy is defined by a proxy protocol, a proxy URI (or URI template
+{{!URITEMPLATE=RFC6570}}), along with potentially other keys.
+
+This document defines two mandatory keys for the sub-dictionaries in the
+`proxies` array, `protocol` and `uri`. These keys, defined below, are
+the initial contents of the proxy information key registry
+({{proxy-info-iana}}). Other optional keys can be added to the dictionary
+to further define or restrict the use of a proxy. Clients that do not
+recognize or understand a key in a proxy sub-dictionary MUST ignore the entire
+proxy definition, since the proxy might be only applicable for particular
+uses.
+
+| JSON Key | Description | Type | Example |
+| --- | --- | --- | --- |
+| protocol | The protocol used to communicate with the proxy | String | "connect-udp" |
+| uri | URI or URI template of the proxy | String | "https://proxy.example.org:4443/masque{?target_host,target_port}" |
+
+The values for the `protocol` key are defined in the proxy protocol
+registry ({{proxy-protocol-iana}}). For proxy types that use HTTP Upgrade Tokens (and
+use the `:protocol` pseudo-header), the `protocol` value will match the Upgrade Token
+/ `:protocol` value.
+
+| Proxy Protocol | Reference |
+| --- | --- |
+| socks | {{!SOCKS=RFC1928}} |
```

Comment:
Since some stacks still support SOCKSv4, and there might be a SOCKSv6, maybe add the version?
```suggestion
| socks5 | {{!SOCKSv5=RFC1928}} |
```